paddle/fluid/operators/detail/grpc_server.cc

```diff
@@ -186,19 +201,22 @@ class RequestPrefetch final : public RequestBase {
     InitializeVariable(var, var_desc->GetType());
     executor_->RunPreparedContext(prefetch_ctx_, scope_);
 
-    SerializeToByteBuffer(var_name, var, *dev_ctx_, &reply);
+    SerializeToByteBuffer(var_name, var, *dev_ctx_, &reply_);
 
-    responder_.Finish(reply, ::grpc::Status::OK, this);
     status_ = FINISH;
+    responder_.Finish(reply_, ::grpc::Status::OK,
+                      reinterpret_cast<void*>(static_cast<intptr_t>(req_id_)));
   }
 
  protected:
   std::shared_ptr<VariableResponse> request_;
+  ::grpc::ByteBuffer reply_;
```

Comment:
`reply` need not be a member.

Reply:
I suspect that making replay_ a member fix a bug. ServerAsyncResponseWriter takes a reference. replay will delete before the ServerAsyncResponseWriter actually uses it asynchronously.